ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +407,188 @@ def test_trustdomain_disable_disables_subdomain(self):
+            fp.write(ssh_key_pair[0])
+
+        keys_opts = ' '.join(['--ssh "{}"'.format(k) for k in pub_keys])
+        cmd = 'ipa user-mod {} {}'.format(username, keys_opts)
+        master.run_command(cmd)
+
+        # connect with first SSH key
+        user_ssh_priv_key = os.path.join(tmpdir, 'ssh_priv_key')
+
+        # change private key permission to comply with SS rules
+        os.chmod(user_ssh_priv_key, 0o600)
+
+        ssh_key_file_contents = ''
+        with open(user_ssh_priv_key, 'r') as infile:
+            ssh_key_file_contents = infile.read()
+        master.put_file_contents('/tmp/user_ssh_priv_key',
+                                 ssh_key_file_contents)
+        master.run_command('chmod 600 /tmp/user_ssh_priv_key')
+
+        # add group parent
+        cmd_output = master.run_command(['ipa', 'group-add', 'parent'])
+        assert 'Added group "parent"' in cmd_output.stdout_text
+
+        # add group child1
+        cmd_output = master.run_command(['ipa', 'group-add', 'child1'])
+        assert 'Added group "child1"' in cmd_output.stdout_text
+
+        # add group child2
+        cmd_output = master.run_command(['ipa', 'group-add', 'child2'])
+        assert 'Added group "child2"' in cmd_output.stdout_text
+
+        # add group members
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'child1', '--groups', 'parent'])
+        assert 'Group name: child1' in cmd_output.stdout_text
+        assert 'Member groups: parent' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'child2', '--groups', 'child1'])
+        assert 'Group name: child2' in cmd_output.stdout_text
+        assert 'Member groups: child1' in cmd_output.stdout_text
+        assert 'Indirect Member groups: parent' in cmd_output.stdout_text
+
+        # add user to group 'parent'
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'parent', '--users', username])
+        assert 'Group name: parent' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member of groups: child1' in cmd_output.stdout_text
+        assert 'Indirect Member of group: child2' in cmd_output.stdout_text
+
+        # clear sssd_cache
+        clear_sssd_cache(master)
+
+        # user lookup
+        cmd_output = master.run_command(['ipa', 'user-show', username])
+        assert 'parent' in cmd_output.stdout_text
+        assert 'ipausers' in cmd_output.stdout_text
+        assert 'child1' in cmd_output.stdout_text
+        assert 'child2' in cmd_output.stdout_text
+
+        clear_sssd_cache(master)
+
+        sshcon = paramiko.SSHClient()
+        sshcon.set_missing_host_key_policy(paramiko.AutoAddPolicy())
```

Comment:
Are we even using the paramiko library now?  I see no further reference to `sshcon`.

Is it the case the the ssh connections are being made from client to server, and not directly from this test code,
via `run_command` on line 553?  If so, we can remove the paramiko import and all references to paramiko.
If so we can

Reply:
Fixed it. Forgot to remove those two lines.